python-avd/pyavd/_eos_designs/structured_config/core_interfaces_and_l3_edge/utils.py

```diff
@@ -304,3 +303,14 @@ def _update_common_interface_cfg(
             if p2p_link.include_in_underlay_protocol is True and self.shared_utils.underlay_ldp and default(p2p_link.mpls_ldp, True):  # noqa: FBT003
                 interface.mpls.ldp.interface = True
                 interface.mpls.ldp.igp_sync = True
+
+    def _get_channel_id(self: AvdStructuredConfigCoreInterfacesAndL3EdgeProtocol, p2p_link: T_P2pLinksItem, node_data: dict) -> int:
+        """Returns a channel ID for one p2p_link."""
+        if node_data.channel_id:
+            return node_data.channel_id
+        if p2p_link.port_channel.channel_id_generation == "p2p_link_id" and p2p_link.id:
```

Comment:
I think we should raise in the case of p2p_link.id not being set but the chosen algorithm being "p2p_link_id".

Reply:
raised error for this, thanks!